### PR TITLE
Ongoing cmake build work.

### PR DESCRIPTION
**Pull request by vargaz**
* Fix indentation.
* Add linux support.
* Add tvos support.
* Use clang only options only with clang.
* Add check for iconv.

### CODE REVIEW

File: src/mono/CMakeLists.txt [hunk 1]
```diff
@@ -164,16 +165,24 @@ if(GCC)
...
+  set(WARNINGS "-Wall -Wunused -Wmissing-declarations -Wpointer-arith -Wno-cast-qual -Wwrite-strings -Wno-switch -Wno-switch-enum -Wno-unused-value -Wno-attributes -Wmissing-prototypes -Wstrict-prototypes -Wnested-externs -Wno-format-zero-length -Wno-unused-function")
 
-  set(WERROR "-Werror=incompatible-pointer-types -Werror=return-type -Werror-implicit-function-declaration")
+  if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+    set(WARNINGS "${WARNINGS} -Qunused-arguments -Wno-tautological-compare -Wno-parentheses-equality -Wno-self-assign -Wno-return-stack-address -Wno-constant-logical-operand -Wno-zero-length-array")
+  endif()
+
+  check_c_compiler_flag("-Werror=incompatible-pointer-types" WERROR_INCOMPATIBLE_POINTER_TYPES)
+  if(WERROR_INCOMPATIBLE_POINTER_TYPES)
+	set(WERROR "${WERROR} -Werror=incompatible-pointer-types")
```
Comment (CoffeeFlux) [inline src/mono/CMakeLists.txt:176]: This is a tab, which I assume is a mistake? Also, why are we using spaces here—is this a dotnet/runtime CMake standard?

File: src/mono/CMakeLists.txt [hunk 2]
```diff
@@ -202,16 +211,37 @@ elseif(CMAKE_SYSTEM_NAME STREQUAL "iOS")
   set(DISABLE_CRASH_REPORTING 1)
   set(ENABLE_MONOTOUCH 1)
   add_definitions(-DMONOTOUCH=1)
-  add_definitions(-DHOST_IOS=1)
   set(HAVE_SYSTEM 0)
   set(MONO_DL_NEED_USCORE 1)
+  if(CMAKE_SYSTEM_NAME STREQUAL "tvOS")
+	set(HOST_TVOS 1)
```
Comment (CoffeeFlux) [inline src/mono/CMakeLists.txt:217]: Tab again. Maybe specify the cmake files in the editorconfig as well, as a followup to Aleksey's PR?

File: src/mono/CMakeLists.txt [hunk 3]
```diff
@@ -230,9 +260,10 @@ endif()
 if(TARGET_SYSTEM_NAME STREQUAL "Darwin")
   set(TARGET_MACH 1)
   set(TARGET_OSX 1)
-elseif(TARGET_SYSTEM_NAME STREQUAL "iOS")
+elseif(TARGET_SYSTEM_NAME STREQUAL "iOS" OR TARGET_SYSTEM_NAME STREQUAL "tvOS")
   set(TARGET_MACH 1)
   set(TARGET_IOS 1)
+elseif(TARGET_SYSTEM_NAME STREQUAL "Linux")
```
Comment (CoffeeFlux) [inline src/mono/CMakeLists.txt:266]: Nothing in the block, is this needed?

File: src/mono/CMakeLists.txt [hunk 4]
```diff
@@ -259,10 +290,9 @@ elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86")
   set(HOST_X86 1)
 elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "arm64")
   set(HOST_ARM64 1)
-elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "arm")
+elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "arm" OR CMAKE_SYSTEM_PROCESSOR STREQUAL "armv7s")
```
Comment (CoffeeFlux) [inline src/mono/CMakeLists.txt:293]: Similarly, isn't `arm64e` an option needed next to `arm64`?

File: src/mono/cmake/configure.cmake [hunk 5]
```diff
@@ -100,3 +100,16 @@ check_type_size("void*" SIZEOF_VOID_P)
...
+find_library(LIBICONV_FOUND iconv)
+if(NOT LIBICONV_FOUND STREQUAL "LIBICONV_FOUND-NOTFOUND")
+  set(ICONV_LIB "iconv")
+endif()
+
+file(WRITE ${CMAKE_BINARY_DIR}{$CMAKE_FILES_DIRECTORY}/CMakeTmp/test.c
+  "#include <sched.h>\n"
+  "void main () { CPU_COUNT((void *) 0); }\n"
+)
+try_compile(GLIBC_HAS_CPU_COUNT ${CMAKE_BINARY_DIR}/CMakeTmp SOURCES "${CMAKE_BINARY_DIR}{$CMAKE_FILES_DIRECTORY}/CMakeTmp/test.c"
```
Comment (CoffeeFlux) [inline src/mono/cmake/configure.cmake:114]: CMake path handling is so bad :(
I think `cmake_path` got merged upstream earlier this month? But it's still clunky and we probably can't depend on a bleeding edge version, alas.

**Review by CoffeeFlux — APPROVED** [review]

File: src/mono/mono/mini/simd-intrinsics-netcore.c [hunk 6]
```diff
@@ -2181,7 +2181,9 @@ MONO_EMPTY_SOURCE_FILE (simd_intrinsics_netcore);
 void
 ves_icall_System_Runtime_Intrinsics_X86_X86Base___cpuidex (int abcd[4], int function_id, int subfunction_id)
 {
+#ifndef MONO_CROSS_COMPILE
```
Comment (CoffeeFlux) [inline src/mono/mono/mini/simd-intrinsics-netcore.c:2184]: How does this affect the autotools builds? They'll never define `MONO_CROSS_COMPILE`, right?

Comment (vargaz) [reply src/mono/mono/mini/simd-intrinsics-netcore.c:2184]: This will only affect cross compilers.

File: src/mono/CMakeLists.txt (hunk 1 above)
Comment (vargaz) [reply src/mono/CMakeLists.txt:176]: Not sure whenever we should use spaces or tabs.

**Pull request merged**